sympy/simplify/simplify.py

```diff
@@ -1569,32 +1571,64 @@ def _reduce_inverses(_args):
         inverses = []
         args = []
         for arg in _args:
-            if isinstance(arg, Pow) and arg.args[1] < 0:
+            if isinstance(arg, _Pow) and arg.args[1] < 0:
                 inverses = [arg**-1] + inverses
                 inv_tot += 1
             else:
-                if inverses:
-                    args.append(Pow(Mul(*inverses), -1))
+                if len(inverses) == 1:
+                    args.append(inverses[0]**-1)
+                elif len(inverses) > 1:
+                    args.append(_Pow(_Mul(*inverses), -1))
                     inv_tot -= len(inverses) - 1
                 inverses = []
                 args.append(arg)
         if inverses:
-            args.append(Pow(Mul(*inverses), -1))
+            args.append(_Pow(_Mul(*inverses), -1))
             inv_tot -= len(inverses) - 1
         return inv_tot, tuple(args)
-    # ========================================================
 
+    def get_score(s):
+        # compute the number of arguments of s
+        # (including in nested expressions) overall
+        # but ignore exponents
+        if isinstance(s, _Pow):
+            return get_score(s.args[0])
+        elif isinstance(s, (_Add, _Mul)):
+            return sum([get_score(a) for a in s.args])
+        return 1
+
+    def compare(s, alt_s):
+        # compare two possible simplifications and return a
+        # "better" one
+        if s != alt_s and get_score(alt_s) < get_score(s):
+            return alt_s
+        return s
+    # ========================================================
 
-    if not isinstance(expr, (Add, Mul, Pow)):
+    if isinstance(expr, MatrixExpr):
+        expr = expr.doit(inv_expand=False)
+        _Add, _Mul, _Pow, _Symbol = MatAdd, MatMul, MatPow, MatrixSymbol
+    else:
+        _Add, _Mul, _Pow, _Symbol = Add, Mul, Pow, Symbol
+    if not isinstance(expr, (_Add, _Mul, _Pow)) or expr.is_commutative:
         return expr
     args = expr.args[:]
-    if isinstance(expr, Pow):
+    if isinstance(expr, _Pow):
         if deep:
-            return Pow(nc_simplify(args[0]), args[1])
+            return _Pow(nc_simplify(args[0]), args[1]).doit()
         else:
             return expr
-    elif isinstance(expr, Add):
-        return Add(*[nc_simplify(a, deep=deep) for a in args])
+    elif isinstance(expr, _Add):
+        return _Add(*[nc_simplify(a, deep=deep) for a in args]).doit()
+    else:
+        # get the non-commutative part
+        com_coeff = 1
+        i = 0
+        while args[i].is_commutative:
```

Comment:
Can you use expr.args_cnc here?

Reply:
_That's_ what it was called. I thought there was a function like this but couldn't find the name anywhere... I'll check if it works properly on matrix expressions but yes, that would probably be better.